src/app/visual/dock/dock-visual.component.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  AfterViewInit,
+  Component
+} from '@angular/core';
+
+import {
+  SkyDockService
+} from '../../public';
+
+import {
+  DockItemVisualContext
+} from './dock-item-context';
+
+import {
+  DockItemVisualComponent
+} from './dock-item-visual.component';
+
+@Component({
+  selector: 'dock-visual',
+  templateUrl: './dock-visual.component.html',
+  styleUrls: ['./dock-visual.component.scss']
+})
+export class DockVisualComponent implements AfterViewInit {
+
+  public stackOrder: number;
+
+  private configs: any[] = [
+    {
+      stackOrder: 0,
+      backgroundColor: 'darkred'
+    },
+    {
+      stackOrder: 100,
+      backgroundColor: 'darkmagenta'
+    },
+    {
+      stackOrder: 10,
+      backgroundColor: 'darkcyan'
+    },
+    {
+      stackOrder: -1000,
+      backgroundColor: 'darkblue'
+    },
+    {
+      stackOrder: 1,
+      backgroundColor: 'darkgreen'
+    }
+  ];
+
+  constructor(
+    private dockService: SkyDockService
+  ) { }
+
+  public ngAfterViewInit(): void {
+    setTimeout(() => {
```

Comment:
Creating dynamic components must not happen within a lifecycle hook.
https://www.reddit.com/r/Angular2/comments/837tuk/getting_an/?utm_source=share&utm_medium=web2x

There are three options:
- Call the service in the component's constructor, before the first "tick" (I modified the visual test to do this).
- Call the service in a component's lifecycle hook (`ngOnInit`, etc) but wrap it in a `setTimeout` so that the component creation happens during the next "tick".
- Move the `setTimeout` to the dock service, making the method call asynchronous. I didn't want to do this because calls to the service outside of a lifecycle hook work perfectly (e.g., via a button click, etc.) so it felt needlessly complicated to make the method asynchronous just for a few edge cases.

We can document this in the code examples.

(Does this explanation work for you, or am I missing something critical?)

Reply:
As long as its documented, then I'm good with it! Just curious - is there a way to throw a console warning if its not used in the constructor? That might be tough... not a deal breaker, just a thought.

Reply:
I don't think so, unfortunately. This will be a problem for any service that creates dynamic components within a lifecycle hook. Just something we'll need to be aware of and document correctly in our code examples.